.azure-pipelines/pull_request.yml

```diff
@@ -9,116 +9,17 @@ pr:
-              # TODO: source activate does not work (see https://github.com/conda/conda/issues/9566)
-              echo "##vso[task.prependpath]$CONDA/envs/scipp-developer/bin"
-              mkdir -p  "$(build_dir)" "$(install_dir)" "$(docs_build_dir)"
-            displayName: 'Configure build environment'
-          - bash: |
-              set -ex
-              cd "$(build_dir)"
-              cmake \
-                -G Ninja \
-                -DPYTHON_EXECUTABLE=$(command -v python) \
-                -DCMAKE_INSTALL_PREFIX="$(install_dir)" \
-                -DCMAKE_BUILD_TYPE=Release \
-                -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=OFF \
-                -DDYNAMIC_LIB=ON \
-                "$(Build.Repository.LocalPath)"
-            displayName: 'CMake configuration'
-          - bash: |
-              set -ex
-              cd "$(build_dir)"
-              cmake --build . --target all-tests
-            displayName: 'Build C++ unit tests'
-          - bash: |
-              set -ex
-              cd "$(build_dir)"
-              ./bin/scipp-common-test
-              ./bin/scipp-units-test
-              ./bin/scipp-core-test
-              ./bin/scipp-variable-test
-              ./bin/scipp-dataset-test
-            displayName: 'Run C++ unit tests'
-          - bash: |
-              set -ex
-              cd "$(build_dir)"
-              cmake --build . --target all-benchmarks
-            displayName: 'Build benchmarks'
-          - bash: |
-              set -ex
-              cd "$(build_dir)"
-              cmake --build . --target install
-            displayName: 'Build Python library'
-          - bash: |
-              set -ex
-              cd package/scipp-package-test
-              mkdir -p build && cd build
-              cmake -DCMAKE_PREFIX_PATH="$(install_dir)" ..
-              cmake --build .
-              ./test
-            displayName: 'Check packaging'
-          - bash: |
-              set -ex
-              PYTHONPATH="$PYTHONPATH:$(install_dir)" python -m pytest
-            displayName: 'Run Python unit tests'
-          - bash: |
-              set -ex
-              cd docs
-              # Build documentation, redirecting doctrees to avoid size bloat in build documentation
-              PYTHONPATH="$PYTHONPATH:$(install_dir)" sphinx-build -d $(mktemp -d) . "$(docs_build_dir)"
-              # Remove Juptyer notebooks used for documentation build, they are not accessible and create size bloat
-              find "$(docs_build_dir)" -type f -name *.ipynb -delete
-            displayName: 'Build documentation'
-          - task: PublishBuildArtifacts@1
-            inputs:
-              PathtoPublish: "$(docs_build_dir)"
-              ArtifactName: 'documentation'
-            displayName: 'Upload built documentation'
-          - bash: ccache --show-stats
-            displayName: 'Report ccache statistics'
+      - template: templates/build_linux.yml
```

Comment:
Does it make sense to move the Mac and Windows builds into template files, too?

Reply:
Yes, that's the goal. Please hold off with revieweing for now. PR is opened only so I can try the azure builds.